src/server/middlewares/auth.ts

```diff
@@ -0,0 +1,75 @@
+import jwt from "jsonwebtoken"
+import { Request, Response, NextFunction } from "express"
+import config from "@config"
+import { JwtPayload } from "../../domain/shared/primitives"
+import { Account } from "../../models/account"
+import LoggerInstance from "../../server/loaders/logger"
+
+interface VerifyTokenRes {
+  success: boolean
+  msg?: string
+  payload?: JwtPayload
+}
+
+function verifyToken(req: Request): VerifyTokenRes {
+  LoggerInstance.info("Verifying Auth token: %o")
+  try {
+    const token = req.headers.authorization?.split(" ")[1]
+    if (!token) {
+      return { success: true, msg: "invalid token" }
+    }
+    const payload = jwt.verify(token, config.secret) as JwtPayload
+    return { success: true, payload }
+  } catch (error) {
+    LoggerInstance.error("Error Verifying Auth token: %o", error)
+    return { success: false, msg: "invalid token" }
+  }
+}
+
+export async function customerAuth(req: Request, res: Response, next: NextFunction) {
+  LoggerInstance.info("Verifying Customer Auth token: %o")
+  try {
+    const verify: VerifyTokenRes = verifyToken(req)
+    if (!verify.success) {
+      return res.status(401).json({ success: false, message: verify.msg })
+    }
+    if (verify.payload?.admin) {
+      return res.status(401).json({ success: false, message: "unauthorize" })
+    }
+    const user = (await Account.findOne({
```

Comment:
finding an account should be the service layer. You can create an `AccountRepository` function in `services/Account` that receives an `id` and returns the user account.